File: moderate/minimum_coins/minimum.coins.cpp
// Reuben Thorpe (2016), CodeEval [Minimum Coins v1.1]
#include <iostream>
#include <fstream>

using namespace std;


int main(int argc, char* argv[]) {
  ifstream in_file(argv[1]);
  int num;

  while (in_file >> num) {
    int num_coins;
    num_coins = num / 5;
    num %= 5;
    num_coins += num / 3;
    num %= 3;
    num_coins += num;
    cout << num_coins << '\n';
  }

  return(0);
}

File: 02-moderate/minimum_coins/minimum.coins.py
# Reuben Thorpe (2016), CodeEval [Minimum Coins v1.0]
from sys import argv


def min_coins(num):
    num_coins = num // 5
    num %= 5
    num_coins += num // 3
    num %= 3
    num_coins += num
    return(num_coins)


if __name__ == "__main__":
    for num in open(argv[1], "r"):
        print(min_coins(int(num)))
